src/hash_table.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import annotations
+from typing import Union
+import random
+
+
+class Node:
+    def __init__(self, key: str, val: Union[str, float], next: Node = None):
+        self.key = key
+        self.val = val
+        self.next = next
+
+
+class ChainHashTable:
```

Comment:
Did you forget making `Node` be a private class?

Reply:
I was waiting for QA's comment. Gonna update at the next round of commits.